src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -3221,6 +3221,9 @@ void visitArgumentEvaluateAndUnsplit(int argumentIndex, bool assertsTrue, bool a
                         var parameterValue = new BoundParameter(argument.Syntax, parameter);
                         var lValueType = result.LValueType;
                         TrackNullableStateForAssignment(parameterValue, lValueType, MakeSlot(argument), parameterWithState);
+
+                        // check whether parameter would unsafely let a null out
+                        ReportNullableAssignmentIfNecessary(parameterValue, lValueType, parameterWithState, useLegacyWarnings: false);
```

Comment:
>ReportNullableAssignmentIfNecessary(parameterValue, lValueType, parameterWithState, useLegacyWarnings: false); [](start = 24, length = 110)

Do we want to report warnings even when suppressed?

Reply:
We should probably confirm, but I think so.
The suppression operator does two things: it changes the nullable state of an expression to non-null and it suppresses conversion warnings.

Reply:
Opened https://github.com/dotnet/roslyn/issues/35555 for discussion

Reply:
Turns out my reading of the code was wrong. We have kept the original behavior (and it's covered by ` RefParameter_TopLevelNullability_Suppressed`). 
If the expression is suppressed, we get a non-null state on the way out, thus no warning from assignment.